components/calendar/src/chinese_based.rs

```diff
@@ -0,0 +1,331 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! This module contains types and traits for use in the Chinese traditional lunar calendar,
```

Comment:
Currently having an issue with this; here is the rustdoc I wrote:
```rust
//! use icu::calendar::{chinese::Chinese, Iso, Date};
//!
//! let iso_date = Date::try_new_iso_date(2023, 6, 23).unwrap();
//! let chinese_date = Date::new_from_iso(iso_date, Chinese);
//! // TODO: When Korean and Vietnamese are added, put example code here as well.
//!
//! assert_eq!(chinese_date.year().number, 4660);
//! assert_eq!(chinese_date.year().related_iso, Some(2023));
//! assert_eq!(chinese_date.year().cyclic, Some(40));
//! assert_eq!(chinese_date.month().ordinal, 6);
//! assert_eq!(chinese_date.day_of_month().0, 6);
```

And the error I get when I try to doctest: 
```
error[E0423]: expected value, found struct `Chinese`
 --> src/chinese_based.rs:12:49
  |
6 | let chinese_date = Date::new_from_iso(iso_date, Chinese);
  |                                                 ^^^^^^^ constructor is not visible here due to private fields
  |
help: consider importing one of these items instead
  |
2 + use icu_calendar::AnyCalendar::Chinese;
  |
2 + use icu_calendar::AnyCalendarKind::Chinese;
  |
2 + use icu_calendar::any_calendar::AnyDateInner::Chinese;
  |
```

Reply:
I want to clarify that importing `icu_calendar::AnyCalendar::Chinese` doesn't work either. Also, this is only a problem in this specific doctest; my unit tests do not have this issue, it's only arising now. 

It's also worth noting that `chinese_based` is not a public facing module; all visible code is in implementing calendars like `Chinese`, so it might not even be necessary to have rustdoc code here, although it may be helpful. But really, any rustdoc code I write here is pretty much the same as what would be written in `Chinese` or `Korean`/`Vietnamese` when they are implemented.

Reply:
Ah, this is happening because `Chinese` is marked as `#[non_exhaustive]`. You should apply `#[allow(clippy::exhaustive_structs)] // this type is stable` to the type instead of `#[non_exhaustive]`.

Reply:
This is resolved now.